app/src/main/kotlin/com/wire/android/workmanager/worker/PersistentWebsocketCheckWorker.kt

```diff
@@ -0,0 +1,111 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+package com.wire.android.workmanager.worker
+
+import android.content.Context
+import android.os.Build
+import androidx.hilt.work.HiltWorker
+import androidx.work.BackoffPolicy
+import androidx.work.Constraints
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkRequest.Companion.MIN_BACKOFF_MILLIS
+import androidx.work.WorkerParameters
+import com.wire.android.appLogger
+import com.wire.android.feature.ShouldStartPersistentWebSocketServiceUseCase
+import com.wire.android.services.PersistentWebSocketService
+import com.wire.android.workmanager.worker.PersistentWebsocketCheckWorker.Companion.NAME
+import com.wire.android.workmanager.worker.PersistentWebsocketCheckWorker.Companion.TAG
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.coroutineScope
+import java.util.concurrent.TimeUnit
+
+@HiltWorker
+class PersistentWebsocketCheckWorker
+@AssistedInject constructor(
+    @Assisted private val appContext: Context,
+    @Assisted private val workerParams: WorkerParameters,
+    private val shouldStartPersistentWebSocketService: ShouldStartPersistentWebSocketServiceUseCase
+) : CoroutineWorker(appContext, workerParams) {
+
+    override suspend fun doWork(): Result = coroutineScope {
+        appLogger.i("Periodic check for persistent websocket connection")
+        val persistentWebSocketServiceIntent = PersistentWebSocketService.newIntent(appContext)
+        // todo, move this logic out, maybe to a separate use case and reuse in the boot broadcast receiver.
+        shouldStartPersistentWebSocketService().let {
+            when (it) {
+                is ShouldStartPersistentWebSocketServiceUseCase.Result.Failure -> {
+                    appLogger.e("${TAG}: Failure while fetching persistent web socket status flow")
+                }
+
+                is ShouldStartPersistentWebSocketServiceUseCase.Result.Success -> {
+                    if (it.shouldStartPersistentWebSocketService) {
+                        if (PersistentWebSocketService.isServiceStarted) {
+                            appLogger.i("${TAG}: PersistentWebsocketService already started, not starting again")
+                        } else {
+                            appLogger.i("${TAG}: Starting PersistentWebsocketService")
+                            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                                appContext.startForegroundService(persistentWebSocketServiceIntent)
+                            } else {
+                                appContext.startService(persistentWebSocketServiceIntent)
+                            }
```

Comment:
this way of startingthe service wi lead to a crash
you can use
``` kotlin
        ServiceCompat.startForeground(this, PERSISTENT_NOTIFICATION_ID, notification, ServiceInfo.FOREGROUND_SERVICE_TYPE_SPECIAL_USE)
``` 
and it does not require checking the android SDK version

Reply:
wait by bad this is a different thing pleases ignore